flocker/node/script.py

```diff
@@ -304,6 +264,17 @@ def flocker_dataset_agent_main():
     loopback block device backend.  Later it will be capable of starting a
     dataset agent using any of the support dataset backends.
     """
+    # XXX This should use dynamic dispatch in the deployer_factory
+    # There should be only AgentScript, not ZFSAgentScript, and it should
+    # do the right thing for the configured backend. FLOC-1791.
+
+    options = DatasetAgentOptions()
+
+    return FlockerScriptRunner(
```

Comment:
Are you sure? 

```
setup.py
169:            'flocker-dataset-agent = flocker.node.script:flocker_dataset_agent_main',  # noqa

```

And

```
flocker/node/functional/test_script.py
10:class FlockerDatasetAgentTests(make_script_tests(b"flocker-dataset-agent")):
12:    Tests for ``flocker-dataset-agent``.

```

Or do you mean that it will now be subject to further tests....tests that had previously launched `flocker-zfs-agent`.

...I'll read on.